Add device_check place holder for functions

wenleix opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56872 Support auto generation of device check
* #56871 Disable device check for foreach kernels
* **#56870 Add device_check place holder for functions**
* #56869 Fix tensor device in test_kthvalue_overlap

Automatic generation of device check code will be supported in
following PRs.

Changed are generaetd via:

1. Codemod
```
fastmod '  device_guard: False' '  device_check: NoCheck
  device_guard: False' aten/src/ATen/native/native_functions.yaml
```

2. Python script: https://gist.github.com/wenleix/be20c34bbbfcee0b289cdea2cf15b16c

Differential Revision: [D27986427](https://our.internmc.facebook.com/intern/diff/D27986427/)

Commit:
Add device_check place holder for functions

Automatic generation of device check code will be supported in
following PRs.

Changed are generaetd via:

1. Codemod
```
fastmod '  device_guard: False' '  device_check: NoCheck
  device_guard: False' aten/src/ATen/native/native_functions.yaml
```

2. Python script: https://gist.github.com/wenleix/be20c34bbbfcee0b289cdea2cf15b16c

Differential Revision: [D27986427](https://our.internmc.facebook.com/intern/diff/D27986427/)

[ghstack-poisoned]

Commit:
rebase && fix minor issues on "Add device_check place holder for functions"


Automatic generation of device check code will be supported in
following PRs.

Changed are generaetd via:

1. Codemod
```
fastmod '  device_guard: False' '  device_check: NoCheck
  device_guard: False' aten/src/ATen/native/native_functions.yaml
```

2. Python script: https://gist.github.com/wenleix/be20c34bbbfcee0b289cdea2cf15b16c

Differential Revision: [D27986427](https://our.internmc.facebook.com/intern/diff/D27986427/)

[ghstack-poisoned]

Commit:
fiix copysign&sparse on "Add device_check place holder for functions"


Automatic generation of device check code will be supported in
following PRs.

Changed are generaetd via:

1. Codemod
```
fastmod '  device_guard: False' '  device_check: NoCheck
  device_guard: False' aten/src/ATen/native/native_functions.yaml
```

2. Python script: https://gist.github.com/wenleix/be20c34bbbfcee0b289cdea2cf15b16c

Differential Revision: [D27986427](https://our.internmc.facebook.com/intern/diff/D27986427/)

[ghstack-poisoned]

Commit:
fix cudnn&&nonzero on "Add device_check place holder for functions"


Automatic generation of device check code will be supported in
following PRs.

Changed are generaetd via:

1. Codemod
```
fastmod '  device_guard: False' '  device_check: NoCheck
  device_guard: False' aten/src/ATen/native/native_functions.yaml
```

2. Python script: https://gist.github.com/wenleix/be20c34bbbfcee0b289cdea2cf15b16c

Differential Revision: [D27986427](https://our.internmc.facebook.com/intern/diff/D27986427/)

[ghstack-poisoned]

ezyang commented on tools/codegen/model.py:
```diff
@@ -291,6 +291,8 @@ def from_yaml(ei: Dict[str, object], loc: 'Location') -> 'NativeFunction':
         device_guard = e.pop('device_guard', True)
         assert isinstance(device_guard, bool), f'not a bool: {device_guard}'
 
+        device_check_s = e.pop('device_check', None)
```
...might as well include the parsing code here?

ezyang commented:
Since this PR doesn't make sense without extra follow up PRs, it behooves you to describe how these will get kept up to date, esp as people keep adding new functions, etc.

Commit:
rebase & address comment on "Add device_check place holder for functions"


Automatic generation of device check code will be supported in
following PRs.

Changed are generaetd via:

1. Codemod
```
fastmod '  device_guard: False' '  device_check: NoCheck
  device_guard: False' aten/src/ATen/native/native_functions.yaml
```

2. Python script: https://gist.github.com/wenleix/be20c34bbbfcee0b289cdea2cf15b16c

Differential Revision: [D27986427](https://our.internmc.facebook.com/intern/diff/D27986427/)

[ghstack-poisoned]

Commit:
rebase && fix special xlogy on "Add device_check place holder for functions"


Automatic generation of device check code will be supported in
following PRs.

Changed are generaetd via:

1. Codemod
```
fastmod '  device_guard: False' '  device_check: NoCheck
  device_guard: False' aten/src/ATen/native/native_functions.yaml
```

2. Python script: https://gist.github.com/wenleix/be20c34bbbfcee0b289cdea2cf15b16c

Differential Revision: [D27986427](https://our.internmc.facebook.com/intern/diff/D27986427/)

[ghstack-poisoned]
